process/sync/baseSync.go

```diff
@@ -154,67 +157,49 @@ func (boot *baseBootstrap) processReceivedHeader(headerHandler data.HeaderHandle
 		log.Debug("forkDetector.AddHeader", "error", err.Error())
 	}
 
-	boot.mutRcvHdrHash.Lock()
-	hash := boot.requestedHeaderHash()
-	if hash == nil {
-		boot.mutRcvHdrHash.Unlock()
-		return
-	}
+	go boot.requestMiniBlocks(headerHandler.GetShardID(), headerHandler.GetNonce())
+
+	boot.computeReceivedHeaderByNonce(headerHandler, headerHash)
+	boot.computeReceivedHeaderByHash(headerHandler, headerHash)
+}
 
-	if bytes.Equal(hash, headerHash) {
+func (boot *baseBootstrap) computeReceivedHeaderByNonce(headerHandler data.HeaderHandler, hdrHash []byte) {
```

Comment:
wrong name for this function. It does not compute. It writes on a chan when it had received a header

Reply:
method was renamed.